sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -144,31 +144,29 @@ EventImplPtr Scheduler::addCG(
     }
   }
 
-  enqueueCommandForCG(NewEvent, AuxiliaryCmds);
+  enqueueCommandForCG(*NewEvent, AuxiliaryCmds);
 
   if (!AuxiliaryResources.empty())
     registerAuxiliaryResources(NewEvent, std::move(AuxiliaryResources));
 
   return NewEvent;
 }
 
-void Scheduler::enqueueCommandForCG(EventImplPtr NewEvent,
+void Scheduler::enqueueCommandForCG(event_impl &Event,
                                     std::vector<Command *> &AuxiliaryCmds,
                                     BlockingT Blocking) {
   std::vector<Command *> ToCleanUp;
   {
     ReadLockT Lock = acquireReadLock();
 
-    Command *NewCmd = (NewEvent) ? NewEvent->getCommand() : nullptr;
+    Command *NewCmd = Event.getCommand();
 
     EnqueueResultT Res;
     bool Enqueued;
 
     auto CleanUp = [&]() {
       if (NewCmd && (NewCmd->MDeps.size() == 0 && NewCmd->MUsers.size() == 0)) {
-        if (NewEvent) {
-          NewEvent->setCommand(nullptr);
-        }
+        Event.setCommand(nullptr);
         delete NewCmd;
```

Comment:
This `delete` is just bizarre, the ownership model is totally unclear here. Irrelevant to this PR though.